From 2080fa3c81b3810228ae5a8abbb494c78dd1d566 Mon Sep 17 00:00:00 2001
From: Timo <timo.teras@iki.fi>
Date: Thu, 13 Sep 2018 20:25:33 +0200
Subject: device: Fix wrong mask len

Bug: 14695

The `mask_len` field is setup incorrectly in
`xfwm_device_fill_xi2_event_mask()`.

The size is the pointer length, not the bitmask data area length.
---
 src/device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/device.c b/src/device.c
index b1e33ae2..01c0c81d 100644
--- a/src/device.c
+++ b/src/device.c
@@ -301,7 +301,7 @@ xfwm_device_fill_xi2_event_mask (XIEventMask *xievent_mask, gulong core_mask)
     guint i;
 
     xievent_mask->deviceid = XIAllMasterDevices;
-    xievent_mask->mask_len = sizeof (mask);
+    xievent_mask->mask_len = len;
     xievent_mask->mask = mask;
 
     for (i = 0; i < G_N_ELEMENTS (core_to_xi2); i++)
-- 
cgit v1.2.1

